python-sdk/tests_integration/sql/operators/test_merge.py

```diff
@@ -261,3 +258,166 @@ def test_merge_with_different_schemas(database_table_fixture, multiple_tables_fi
+    "FROM source_table Where true ON CONFLICT (list) DO NOTHING"
+)
+snowflake_single_result_sql = (
+    "merge into IDENTIFIER(:target_table) using IDENTIFIER(:source_table) "
+    "on Identifier(:merge_clause_target_0)=Identifier(:merge_clause_source_0)"
+    " when not matched then insert(target_table.list) values (source_table.list)"
+)
+delta_single_result_sql = (
+    "merge into target_table as `target_table` using source_table as `source_table`"
+    " on `target_table`.`list`=`source_table`.`list` when not matched then "
+    "insert(target_table.list) values (source_table.list)"
+)
+bigquery_single_result_sql = (
+    f"MERGE {SCHEMA}.target_table T USING {SCHEMA}.source_table S ON T.list=S.list "
+    "WHEN NOT MATCHED BY TARGET THEN INSERT (list) VALUES (list)"
+)
+
+base_database_class = "astro.databases.base.BaseDatabase.run_sql"
+delta_database_class = "astro.databases.databricks.delta.DeltaDatabase.run_sql"
+
+
+def get_result_sql(conn_id, mode):
+    if mode == "update":
+        return get_result_sql_update(conn_id)
+    elif mode == "single":
+        return get_result_sql_single(conn_id)
+    return get_result_sql_multi(conn_id)
+
+
+def get_result_sql_update(conn_id):
+    database_type = create_database(conn_id=conn_id).sql_type
+    if database_type == "sqlite":
+        return sqlite_update_result_sql
+    elif database_type == "snowflake":
+        return snowflake_update_result_sql
+    elif database_type == "delta":
+        return delta_update_result_sql
+    elif database_type == "bigquery":
+        return bigquery_update_result_sql
+
+
+def get_result_sql_multi(conn_id):
+    database_type = create_database(conn_id=conn_id).sql_type
+    if database_type == "sqlite":
+        return sqlite_multi_result_sql
+    elif database_type == "snowflake":
+        return snowflake_multi_result_sql
+    elif database_type == "delta":
+        return delta_multi_result_sql
+    elif database_type == "bigquery":
+        return bigquery_multi_result_sql
+
+
+def get_result_sql_single(conn_id):
+    database_type = create_database(conn_id=conn_id).sql_type
+    if database_type == "sqlite":
+        return sqlite_single_result_sql
+    elif database_type == "snowflake":
+        return snowflake_single_result_sql
+    elif database_type == "delta":
+        return delta_single_result_sql
+    elif database_type == "bigquery":
+        return bigquery_single_result_sql
+
+
+@pytest.mark.parametrize(
+    "merge_parameters",
+    [
+        "single",
+        "multi",
+        "update",
+    ],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "database_class,conn_id",
+    [
+        (base_database_class, "sqlite_conn"),
+        (base_database_class, "gcp_conn"),
+        (base_database_class, "snowflake_conn"),
+        (delta_database_class, "databricks_conn"),
+    ],
+    ids=["sqlite", "bigquery", "snowflake", "databricks"],
+)
+def test_merge_sql_generation(database_class, conn_id, merge_parameters):
```

Comment:
Should we move this to `python-sdk/tests/sql/operators/test_merge.py`?

Reply:
@kaxil the only thing here that's not "integration test" is that it does need a real connection ID to create the database. Are you ok with me moving it over considering that?